engine/src/main/java/org/terasology/world/block/BlockRegionIterable.java

```diff
@@ -0,0 +1,134 @@
+package org.terasology.world.block;
+
+import com.google.common.collect.Lists;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Consumer;
+
+public class BlockRegionIterable implements Iterable<Vector3ic> {
+
+    private final BlockRegion region;
+    private List<BlockRegion> subtract;
+
+    private BlockRegionIterable(BlockRegion region, List<BlockRegion> subtract) {
+        this.region = region;
+        this.subtract = subtract;
+    }
+
+    public static BlockRegionIterable.Builder region(BlockRegion region) {
+        return new BlockRegionIterable.Builder(region);
+    }
+
+    @Override
+    public Iterator<Vector3ic> iterator() {
+        return new Iterator<Vector3ic>() {
+            private Vector3i current = null;
+            private final Vector3i next = new Vector3i(region.minX, region.minY, region.minZ);
+
+            public boolean isValid(Vector3ic test) {
+                if (subtract != null) {
+                    for (BlockRegion blockRegion : subtract) {
+                        if (blockRegion.testBlock(test)) {
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+
+            public boolean findNext() {
+                if (current.equals(next)) {
+                    do {
+                        next.z++;
+                        if (next.z >= region.maxZ) {
+                            next.z = region.minZ;
+                            next.y++;
+                            if (next.y >= region.maxY) {
+                                next.y = region.minY;
+                                next.x++;
+                            }
+                        }
+                        if (!region.testBlock(next)) {
+                            return false;
+                        }
+                    } while (!isValid(next));
+                    return true;
+                }
+                return true;
+            }
+
+            @Override
+            public boolean hasNext() {
+                if (current == null) {
+                    return true;
+                }
+
+                if (current.equals(next)) {
+                    return findNext();
+                }
+                return !region.testBlock(next);
+            }
+
+            @Override
+            public Vector3ic next() {
+                if (current == null) {
+                    current = new Vector3i(next);
+                    return next;
+                }
+
+                if (current.equals(next)) {
+                    if (findNext()) {
+                        return next;
+                    }
+                    return null;
+                }
+                current.set(next);
+                return next;
+            }
+        };
+    }
+
+    @Override
+    public void forEach(Consumer<? super Vector3ic> consumer) {
+
```

Comment:
Purposefully empty? Wouldn't it be better to throw something like "UnsupportedOperationException"?